cli/internal/core/engine.go

```diff
@@ -146,75 +146,80 @@ func (e *Engine) generateTaskGraph(pkgs []string, taskNames []string, tasksOnly
 		if err != nil {
 			return err
 		}
-		if !visited.Includes(taskID) {
-			visited.Add(taskID)
-			deps := task.Deps
-
-			if tasksOnly {
-				deps = deps.Filter(func(d interface{}) bool {
-					for _, target := range taskNames {
-						return fmt.Sprintf("%v", d) == target
-					}
-					return false
-				})
-				task.TopoDeps = task.TopoDeps.Filter(func(d interface{}) bool {
-					for _, target := range taskNames {
-						return fmt.Sprintf("%v", d) == target
-					}
-					return false
-				})
-			}
 
-			toTaskID := taskID
-			hasTopoDeps := task.TopoDeps.Len() > 0 && e.TopologicGraph.DownEdges(pkg).Len() > 0
-			hasDeps := deps.Len() > 0
-			hasPackageTaskDeps := false
-			if _, ok := packageTasksDepsMap[toTaskID]; ok {
-				hasPackageTaskDeps = true
-			}
+		// Exit loop if we've already seen this
```

Comment:
This comment makes me think we are `break`ing out of the loop. I think "skip" would be a better word.